src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -0,0 +1,2685 @@
+
+        outputs = self.model(
+            pixel_values,
+            pixel_mask=pixel_mask,
+            decoder_attention_mask=decoder_attention_mask,
+            encoder_outputs=encoder_outputs,
+            inputs_embeds=inputs_embeds,
+            decoder_inputs_embeds=decoder_inputs_embeds,
+            labels=labels,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        denoising_meta_values = (
+            outputs.denoising_meta_values if return_dict else outputs[-1] if self.training else None
+        )
+
+        outputs_class = outputs.intermediate_logits if return_dict else outputs[2]
+        outputs_coord = outputs.intermediate_reference_points if return_dict else outputs[3]
+
+        if self.training and denoising_meta_values is not None:
+            dn_out_coord, outputs_coord = torch.split(outputs_coord, denoising_meta_values["dn_num_split"], dim=2)
+            dn_out_class, outputs_class = torch.split(outputs_class, denoising_meta_values["dn_num_split"], dim=2)
+
+        logits = outputs_class[:, -1]
+        pred_boxes = outputs_coord[:, -1]
+
+        loss, loss_dict, auxiliary_outputs = None, None, None
+        if labels is not None:
+            # First: create the criterion
+            criterion = RTDetrLoss(self.config)
+            criterion.to(self.device)
+            # Third: compute the losses, based on outputs and labels
+            outputs_loss = {}
+            outputs_loss["logits"] = logits
+            outputs_loss["pred_boxes"] = pred_boxes
+            if self.config.auxiliary_loss:
+                enc_topk_logits = outputs.enc_topk_logits if return_dict else outputs[-5]
+                enc_topk_bboxes = outputs.enc_topk_bboxes if return_dict else outputs[-4]
+                auxiliary_outputs = self._set_aux_loss(
+                    outputs_class[:, :-1].transpose(0, 1), outputs_coord[:, :-1].transpose(0, 1)
+                )
+                outputs_loss["auxiliary_outputs"] = auxiliary_outputs
+                outputs_loss["auxiliary_outputs"].extend(self._set_aux_loss([enc_topk_logits], [enc_topk_bboxes]))
+                if self.training and denoising_meta_values is not None:
+                    outputs_loss["dn_auxiliary_outputs"] = self._set_aux_loss(dn_out_class, dn_out_coord)
+                    outputs_loss["denoising_meta_values"] = denoising_meta_values
+
+            loss_dict = criterion(outputs_loss, labels)
+            # Compute total loss, as a weighted sum of the various losses
+            weight_dict = {
+                "loss_vfl": self.config.weight_loss_vfl,
+                "loss_bbox": self.config.weight_loss_bbox,
+                "loss_giou": self.config.weight_loss_giou,
+            }
+            weight_loss_scaled = {k: v * loss_dict[k] for k, v in weight_dict.items()}
+
+            loss = sum(weight_loss_scaled.values())
```

Comment:
@SangbumChoi 

Could you check the loss calculation here?
I suspect that we should not calculate the `weight_loss_scales` dict at all.

Instead we should just do 
```py
loss = sum(loss_dict.values())
```

- As in `RTDetrLoss.forward()` (line 2209 and line 2241), we have already multiply the loss with its weight.
**BTW, I think we are missing loss weight multiplication for aux loss (which should happen after line 2220)**
- To include all aux and dn loss in loss calculation.

Reply:
@jasonkit Thanks for this valuable comment! I think you are right about currently it was kind of weird.

Reply:
https://github.com/huggingface/transformers/pull/29077/commits/2f3d4baced00d3e1858f68414f87316745e83a7c

Reply:
I am try doing model fine tuning with this PR on my own dataset and find that is it not work the same when compare tot he ultralytics's rt-detr implementation.

Reply:
@jasonkit It means working but not the same value right? 

I am also aware of difference in data augmentation in ultralytics implementation. I can share you simple fine-tuning jupyter notebook file if you want

Reply:
@SangbumChoi 
It is indeed not working and that's why I am finding out what might be the cause.

It might be problem of my code as I am wrapping this PR and part of ultralytics code (the augmentation part) with pytorch lightning module and using pytorch lightning's trainer.

I would be helpful if you could share the fine-tuning jupyter notebook!

Reply:
@jasonkit If you leave some email or slack-id I would love to communicate and solve the problem

Reply:
@SangbumChoi 
never mind, I kind of figure out the issue. 
Besides the loss calculation difference, I need to use a lower learning rate and set clip the gradient by norm to 0.1.

BTW, could you help convert and publish the `rtdetr_r101vd` (which is comparable to ultralytics's x size model), I guess my dataset needs that to get desirable accuracy.